src/tests/Regressions/coreclr/GitHub_100536/test100536.csproj

```diff
@@ -0,0 +1,12 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <Optimize>True</Optimize>
```

Comment:
We always build the tests for crossgen testing as checked that has Optimize set to true. It should not be necessary to explicitly set Optimize to true here to get the coverage for the fixed bug.

https://github.com/dotnet/runtime/blob/5c9f64cce935e20603fd101d5d9295e8029021a6/eng/pipelines/coreclr/crossgen2.yml#L53

https://github.com/dotnet/runtime/blob/main/src/tests/Directory.Build.props#L111

Reply:
I can confirm that the crossgen test running mode hits the problem with this regression test even without the Optimize set to true, so it can be removed..